ssl.go

```diff
@@ -71,6 +71,11 @@ func (s *SSL) GetServername() string {
 	return C.GoString(C.SSL_get_servername(s.ssl, C.TLSEXT_NAMETYPE_host_name))
 }
 
+// same as GetSecurityLevel but for use without pointer to SSL
```

Comment:
Question:
I'm not completely following. What is the SSL struct used for?

Reply:
`SSL` (actually `struct ssl_st`) is the main structure, it holds most notably SSL method, the BIOs, the handshake handler, cipher lists, and others. Almost every code that includes openssl/ssl.h starts like this one:

```C
#include <openssl/ssl.h>

int main(int argc, char *argv[]) {
 int rc;
 const SSL_METHOD *method = SSLv23_method();
 SSL_CTX *ctx = SSL_CTX_new(method);
 SSL *ssl = SSL_new(ctx);
 rc = SSL_CTX_set_default_verify_paths(ctx);
 return rc;
}
```

Reply:
But there are cases where we do not have access directly to either `SSL` nor `SSL_CTX`, e.g.: unit tests, where we want to check for the global security level, to enable/disable some tests. In that case we have a call in this commit that initialises the structures to get the level and then free and return.